src/main/java/com/cosmeticos/service/OrderService.java

```diff
@@ -225,9 +230,92 @@ public Order update(OrderRequestBody request) {
 			order.setScheduleId(orderRequest.getScheduleId());
 		}
 
+		//AQUI TRATAMOS O STATUS ACCEPTED QUE VAMOS NA SUPERPAY EFETUAR A RESERVA DO VALOR PARA PAGAMENTO
+		if (orderRequest.getStatus() == Order.Status.ACCEPTED) {
+			this.sendPaymentRequest(orderRequest);
+		}
+
+		//AQUI TRATAMOS O STATUS SCHEDULED QUE VAMOS NA SUPERPAY EFETUAR A RESERVA DO VALOR PARA PAGAMENTO
+		if (orderRequest.getStatus() == Order.Status.SCHEDULED) {
+			this.validateScheduledAndsendPaymentRequest(orderRequest);
+		}
+
 		return orderRepository.save(order);
 	}
 
+	private void validateScheduledAndsendPaymentRequest(Order orderRequest) throws Exception {
+
+		int daysToStart = Integer.parseInt(daysToStartPayment);
+
+		//INSTANCIAMOS O CALENDARIO
+		Calendar c = Calendar.getInstance();
+
+		//DATA ATUAL
+		//EX.: 20/08/2017
+		Date now = c.getTime();
+
+		//PEGAMOS A DATA DE INICIO DO AGENDAMENTO DO PEDIDO
+		Date scheduleDateStart = orderRequest.getScheduleId().getScheduleStart();
+
+		//ATRIBUIMOS A DATA DO AGENDAMENTO DO PEDIDO AO CALENDARIO
+		c.setTime(scheduleDateStart);
+
+		//VOLTAMOS N DIAS, DEFINIDO EM PROPRIEDADES, NO CALENDARIO BASEADO NA DATA DO AGENDAMENTO
+		c.add(Calendar.DATE, -daysToStart);
+
+		//DATA DO AGENDAMENTO MENOS N DIAS NO FORMADO DATE. OU SEJA, A DATA QUE DEVE INICIAR AS TENTAVIDAS DE PAGAMENTO
+		Date dateToStartPayment = c.getTime();
+
+		//SE A DATA ATUAL FOR POSTERIOR A DATA QUE DEVE INICIAR AS TENTATIVAS DE RESERVA DO PAGAMENTO, ENVIAMOS PARA PAGAMENTO
+		if (now.after(dateToStartPayment)) {
+			sendPaymentRequest(orderRequest);
+		}
+	}
+
+	private void sendPaymentRequest(Order orderRequest) throws Exception {
+
+		try {
+			Optional<RetornoTransacao> retornoTransacaoSuperpay = paymentController.sendRequest(orderRequest);
+
+			if(retornoTransacaoSuperpay.isPresent()) {
+
+				//VALIDAMOS SE VEIO O STATUS QUE ESPERAMOS
+				//1 = Pago e Capturado | 2 = Pago e não Capturado (O CORRETO SERIA 1, POIS FAREMOS A CAPTURA POSTERIORMENTE)
+				if(retornoTransacaoSuperpay.get().getStatusTransacao() == 1 ||
+						retornoTransacaoSuperpay.get().getStatusTransacao() == 2) {
+
+					//ENVIAMOS OS DADOS DO PAGAMENTO EFETUADO NA SUPERPAY PARA SALVAR O STATUS DO PAGAMENTO
+					//OBS.: COMO ESSE METODO AINDA NAO FOI IMPLEMENTADO, ELE ESTA RETORNANDO BOOLEAN
+					Boolean updateStatusPagamento = paymentService.updatePaymentStatus(retornoTransacaoSuperpay.get());
+
+					if (updateStatusPagamento == false) {
+						//TODO - NAO SEI QUAL SERIA A MALHOR SOLUCAO QUANDO DER UM ERRO AO ATUALIZAR O STATUS DO PAGAMENTO
```

Comment:
Erros na plataforma deles como: erros 500, connection refused, timeout, service unavailable, essas coisas de infra-estrutura, o ideal seria lançarmos uma exceção nova chamada IntegrationException e capturá-la no OrderController, que por sua vez, retorna ao cliente um 502 (bad gateway).

Se for erro retornado no response deles, teríamos que tratar caso a caso. Retornar http 5xx se o problema for falha no nosso request (tipo usuario preenche td certo mas tem bug do nosso lado que acaba fazendo request errado ao superpay); ou Retornar 4xx caso o usuário esteja enviando dados errados;